lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,501 @@
+        # (dependent on forecast-length)
+        #        Lightning rate thresholds for adjusting the first-guess
+        #        lightning probability (strikes per minute == "min^-1").
+        #        lrt_lev1 must be a function that takes "forecast_period"
+        #        in minutes and returns the lightning rate threshold for
+        #        increasing first-guess lightning probability to risk 1 (LR1).
+        #        This gives a decreasing influence on the extrapolated
+        #        lightning nowcast over forecast_period while retaining an
+        #        influence from the 50 km halo.
+        self.lrt_lev1 = lambda mins: 0.5 + mins * 2. / 360.
+        # Lightning-rate threshold for Lightning Risk 2 level
+        #        lrt_lev2 is the lightning rate threshold (as float) for
+        #        increasing first-guess lightning probability to risk 2 (LR2).
+        self.lrt_lev2 = 0.
+
+        # Set values for handling precipitation rate data
+        #    precipthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip).
+        #        These are the three prob(precip) thresholds and are designed
+        #        to prevent a large probability of lightning being output if
+        #        the probability of precipitation is very low.
+        self.precipthr = (0.0, 0.05, 0.1)
+        #    ltngthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip)
+        #        These are the three prob(lightning) values to scale to.
+        self.ltngthr = (0.0067, 0.25, 1.)
+        #    probability thresholds for increasing the prob(lightning)
+        #        phighthresh for heavy precip (>7mm/hr)
+        #            relates to problightning_values[2]
+        #        ptorrthresh for intense precip (>35mm/hr)
+        #            relates to problightning_values[1]
+        self.phighthresh = 0.4
+        self.ptorrthresh = 0.2
+
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=self.radius,
+            lthru=self.lrt_lev1.__class__, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    @staticmethod
+    def _update_metadata(cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "probability_of_lightning"
```

Comment:
This numbered list is appearing really weirdly in sphinx. I think it may be solved by an extra new line before 1.

Reply:
Blank line added.